app/models/user.rb

```diff
@@ -0,0 +1,53 @@
+class User < ActiveRecord::Base
+
+  has_one :profile
+  has_many :posts
+  has_many :comments
+  has_many :likes
+
+  has_secure_password
+
+  before_create :generate_token
+  after_create :build_profile
+  
+  validates :first_name, :last_name, :presence => true
+  validates :birthdate, :presence => true
+  validates :email, :presence => true, :format => { :with => /@/ }
```

Comment:
I know this was in the reading but you'll want more robust validation.

This regex would not care if a user tried to put in even '@' as their email.